Module not found: Error: Can't resolve '@ember-data/tracking/-private'

https://github.com/rust-lang/crates.io/actions/runs/3408161209/jobs/5668483713

### Reproduction

see https://github.com/rust-lang/crates.io/pull/5277

### Description

It looks like the most recent v4.8 release can't be built with embroider anymore due to the error below:

```
stack: ModuleNotFoundError: Module not found: Error: Can't resolve '@ember-data/tracking/-private' in '$TMPDIR/embroider/c37dd7/node_modules/.pnpm/@ember-data+model@4.8.2_t4t7pte3g4axxazuc5bgp4ddgu/node_modules/@ember-data/model/-private/record-state.js'
    at /home/runner/work/crates.io/crates.io/node_modules/.pnpm/webpack@5.74.0/node_modules/webpack/lib/Compilation.js:2016:28
    at /home/runner/work/crates.io/crates.io/node_modules/.pnpm/webpack@5.74.0/node_modules/webpack/lib/NormalModuleFactory.js:798:13
    at eval (eval at create (/home/runner/work/crates.io/crates.io/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
    at /home/runner/work/crates.io/crates.io/node_modules/.pnpm/webpack@5.74.0/node_modules/webpack/lib/NormalModuleFactory.js:270:22
    at eval (eval at create (/home/runner/work/crates.io/crates.io/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at /home/runner/work/crates.io/crates.io/node_modules/.pnpm/webpack@5.74.0/node_modules/webpack/lib/NormalModuleFactory.js:434:22
    at /home/runner/work/crates.io/crates.io/node_modules/.pnpm/webpack@5.74.0/node_modules/webpack/lib/NormalModuleFactory.js:116:11
    at /home/runner/work/crates.io/crates.io/node_modules/.pnpm/webpack@5.74.0/node_modules/webpack/lib/NormalModuleFactory.js:670:25
    at /home/runner/work/crates.io/crates.io/node_modules/.pnpm/webpack@5.74.0/node_modules/webpack/lib/NormalModuleFactory.js:855:8
    at /home/runner/work/crates.io/crates.io/node_modules/.pnpm/webpack@5.74.0/node_modules/webpack/lib/NormalModuleFactory.js:975:5
    at /home/runner/work/crates.io/crates.io/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:6883:13
    at /home/runner/work/crates.io/crates.io/node_modules/.pnpm/webpack@5.74.0/node_modules/webpack/lib/NormalModuleFactory.js:958:45
 ```

--- a/.gitignore
+++ b/.gitignore
@@ -6,6 +6,7 @@
 concat-stats-for
 dist
 tmp
+packages/tracking/addon
 
 # dependencies
 bower_components

--- a/packages/tracking/addon-main.js
+++ b/packages/tracking/addon-main.js
@@ -1,20 +1,5 @@
-const path = require('path');
-
-const Funnel = require('broccoli-funnel');
-
 module.exports = {
   name: require('./package.json').name,
-
-  options: {
-    babel: {
-      plugins: [require.resolve('./lib/transform-ext.js')],
-    },
-  },
-
-  treeForAddon() {
-    const assetDir = path.join(__dirname, './dist');
-    return this._super.treeForAddon.call(this, new Funnel(assetDir, { include: ['**/*.js'] }));
-  },
 
   treeForVendor() {
     return;

--- a/packages/tracking/package.json
+++ b/packages/tracking/package.json
@@ -22,7 +22,6 @@
     "extends": "../../package.json"
   },
   "dependencies": {
-    "broccoli-funnel": "^3.0.8",
     "ember-cli-babel": "^7.26.11"
   },
   "exports": {
@@ -36,9 +35,9 @@
     "dist"
   ],
   "scripts": {
-    "build": "rollup --config",
+    "build": "rollup --config && babel ./dist --out-dir addon --plugins=./lib/transform-ext.js",
     "start": "rollup --config --watch",
-    "prepack": "rollup --config"
+    "prepack": "pnpm build"
   },
   "ember-addon": {
     "main": "addon-main.js",
@@ -47,6 +46,7 @@
   },
   "devDependencies": {
     "@babel/core": "^7.20.2",
+    "@babel/cli": "^7.19.3",
     "@babel/plugin-proposal-class-properties": "^7.18.6",
     "@babel/plugin-proposal-decorators": "^7.20.2",
     "@babel/plugin-transform-runtime": "^7.19.6",

